mirdata/jams_utils.py

```diff
@@ -0,0 +1,147 @@
+        f0_data=None,
+        section_data=None,
+        multi_section_data=None,
+        key_data=None,
+        artist=None,
+        duration=None,
+        title=None,
+        annotator=None,
+        sandbox=None):
+
+    """Convert annotations to JAMS format.
+
+    Args:
+        beat_data (list or None):
+            A list of tuples of (BeatData, str), where str describes the annotation (e.g. beats_1).
+        chord_data (list or None):
+            A list of tuples of (ChordData, str), where str describes the annotation.
+        note_data (list or None):
+            A list of tuples of (NoteData, str), where str describes the annotation.
+        f0_data (list or None):
+            A list of tuples of (F0Data, str), where str describes the annotation.
+        section_data (list or None):
+            A list of tuples of (SectionData, str), where str describes the annotation.
+        key_data (list or None):
+            A list of tuples of (KeyData, str), where str describes the annotation.
+        artist (str or None):
+            A string describing the artist name.
+        duration (str or None):
+            A float describing the track duration in seconds.
+        title (str or None):
+            A string describing the track title.
+        annotator (dict or None):
+            A dict describing the annotator (e.g. {name: annotator_1}
+        sandbox (dict or None):
+            A dict with extra information related to the track.
+
+    """
+
+    jam = jams.JAMS()
+
+    # metadata
+    if artist is not None:
+        jam.file_metadata.artist = artist
+    if title is not None:
+        jam.file_metadata.title = title
+    if duration is not None:
+        jam.file_metadata.duration = duration
+
+    # beats
+    if beat_data is not None:
+        for beats in beat_data:
+            jam.annotations.append(beats_to_jams(beats))
+
+    # sections
+    if section_data is not None:
+        for sections in section_data:
+            jam.annotations.append(sections_to_jams(sections))
+
+    # multi-sections (sections with multiple levels)
+    if multi_section_data is not None:
+        for sections in multi_section_data:
+            if sections[0][0][0] is not None:
+                jam.annotations.append(multi_sections_to_jams(sections))
+
+    # chords
+    if chord_data is not None:
+        for chords in chord_data:
+            jam.annotations.append(chords_to_jams(chords))
+
+    # keys
+    if key_data is not None:
+        for keys in key_data:
+            jam.annotations.append(keys_to_jams(keys))
+
+    return jam
+
+
+def beats_to_jams(beats):
+    jannot_beat = jams.Annotation(namespace='beat')
+    jannot_beat.annotation_metadata = jams.AnnotationMetadata(data_source='mirdata')
+    for t, p in zip(beats[0].beat_times, beats[0].beat_positions):
+        jannot_beat.append(time=t, duration=0.0, value=p)
+    if beats[1] is not None:
+        jannot_beat.sandbox = jams.Sandbox(name=beats[1])
```

Comment:
maybe this fits better in annotation metadata?

Reply:
you mean as an `AnnotationMetadata` object? I think it doesn't have a `name` field. Otherwise I don't know if `jannot_beat.annotation_metadata = jams.Sandbox(name=beats[1])` makes sense?

Reply:
hmm ok, what if we make it a big more flexible. The annotation metadata object has these fields:
`jams.AnnotationMetadata(curator=None, version='', corpus='', annotator=None, annotation_tools='', annotation_rules='', validation='', data_source='')`

what if, like we're doing for track metadata, the beats tuple is now `(BeatData, dict)` where we treat the keys of the dictionary like we do for the track metadata - assigning them to the fields of the annotation_metadata when they exist, and when they dont assigning them to the sandbox?

Reply:
Same as above - let's punt this for later